src/pages/TransactionDetails.tsx

```diff
@@ -53,6 +55,12 @@ const TransactionDetails = () => {
         }
     }, [primaryWallet, refetch]);
 
+    useEffect(() => {
+        if (isError) {
+            return navigate('/');
```

Comment:
the issue with this approach is that you would always redirect on an error, which is fine for a specific case (e.g. page is open and you switch and it causes an error directly after this action), but if a user would open the page in any other way and an error shows up, it would also cause a redirect without any sort of details